cirq-core/cirq/work/observable_measurement.py

```diff
@@ -466,4 +527,11 @@ def measure_grouped_settings(
             bitstrings = np.logical_xor(flippy_ms.flips, result.measurements['z'])
             accumulator.consume_results(bitstrings.astype(np.uint8, casting='safe'))
 
+        if checkpoint:
+            assert checkpoint_fn is not None, 'mypy'
+            assert checkpoint_other_fn is not None, 'mypy'
+            if os.path.exists(checkpoint_fn):
+                os.rename(checkpoint_fn, checkpoint_other_fn)
+            to_json(list(accumulators.values()), checkpoint_fn)
```

Comment:
This only saves the checkpoint - is loading from a checkpoint an item for a later PR?

Reply:
Added a note to the docstring, but you just call `cirq.read_json` on the file. It will give you the same thing as the return value of this function, namely a list of `BitstringAccumulator`s, which have all the results and can be queried to give you observable values.

Reply:
I added a test that reads in the checkpoints

Reply:
+1 to the checkpoint-reading test, thanks!